webpack/assets/javascripts/react_app/components/HostsIndex/TableRowActions/core.js

```diff
@@ -0,0 +1,45 @@
+import forceSingleton from '../../../common/forceSingleton';
+
+const coreTableRowActionsRegistry = forceSingleton(
+  'coreTableRowActionsRegistry',
+  () => ({})
+);
+
+// Unlike the column registry which is collecting objects that describe table columns,
+// here we collect an object containing a single getActions funtion, which returns an array of kebab action items.
+export const registerGetActions = ({
+  pluginName,
+  getActionsFunc,
+  tableName = 'hosts',
+}) => {
+  if (!coreTableRowActionsRegistry[pluginName])
+    coreTableRowActionsRegistry[pluginName] = {};
+  coreTableRowActionsRegistry[pluginName][tableName] = {
+    getActions: getActionsFunc,
+  };
+};
+
+export const registeredTableRowActions = ({ tableName = 'hosts' }) => {
+  const result = {};
+  Object.keys(coreTableRowActionsRegistry).forEach(pluginName => {
+    if (coreTableRowActionsRegistry[pluginName]?.[tableName]) {
+      result[pluginName] = coreTableRowActionsRegistry[pluginName][tableName];
+    }
+  });
+  // { katello: { getActions: [Function: getActions] } }
```

Comment:
leftovers

Reply:
This one is on purpose, so you know what the object looks like :)